includes/abstracts/abstract-wp-job-manager-form.php

```diff
@@ -136,7 +136,7 @@ public function process() {
 	 * @param array $atts Attributes to use in the view handler.
 	 */
 	public function output( $atts = [] ) {
-		WP_Job_Manager\WP_Job_Manager_Recaptcha::enqueue_scripts();
+		WP_Job_Manager\WP_Job_Manager_Recaptcha::enqueue_recaptcha();
```

Comment:
Haven't tested yet, but it looks fragile to add the hooks that process submit fields in the output and enqueue functions. Those might not get called or get called later than `process()`. Does it only work now because the submit job form has multiple steps?

Since reCAPTCHA is only used on specific forms I think the abstract `WP_Job_Manager_Form` class, which has the common code for every form, isn't the place to set it up. Looks like there are separate `WP_Job_Manager_Form_Edit_Job` and `WP_Job_Manager_Form_Submit_Job` classes, should the hooks be added from here?:

https://github.com/Automattic/WP-Job-Manager/blob/881c4f3d5afc30fcc4c3400b225d5dd2752ef692/includes/forms/class-wp-job-manager-form-submit-job.php#L83

(Unfortunately `WP_Job_Manager_Form_Edit_Job` extends `WP_Job_Manager_Form_Submit_Job` so we still need something to not load it for editing. Maybe a `...Submit_Job::init_recaptcha()` method that the `...Edit_Job` class overrides?)

Reply:
Great point, I also thought about this initially but I ended up taking the quick approach since it worked. I'll look into refactoring

Reply:
Ok so I moved everything inside the recaptcha class. I also did some fixes regarding versioning as it seems that previously the latest version of the paid plugins didn't work with old WPJM versions. I have updated the testing instructions to reflect that.